packages/jellyfish-api-core/__tests__/category/account/listBurnHistory.test.ts

```diff
@@ -0,0 +1,261 @@
+    expect(history[6].amounts[0]).toStrictEqual('10.00000000@DFI')
+  })
+
+  it('should listBurnHistory with filter on txtype None', async () => {
+    const history = await client.account.listBurnHistory({
+      txtype: DfTxType.NONE
+    })
+    expect(history.every(({ type }) => type === 'None')).toBeTruthy()
+  })
+
+  it('should listBurnHistory with filter on txtype UtxosToAccount', async () => {
+    const history = await client.account.listBurnHistory({
+      txtype: DfTxType.UTXOS_TO_ACCOUNT
+    })
+    expect(history.every(({ type }) => type === 'UtxosToAccount')).toBeTruthy()
+  })
+
+  it('should listBurnHistory with filter on txtype AccountToAccount', async () => {
+    const history = await client.account.listBurnHistory({
+      txtype: DfTxType.ACCOUNT_TO_ACCOUNT
+    })
+    expect(history.every(({ type }) => type === 'AccountToAccount')).toBeTruthy()
+  })
+
+  it('should listBurnHistory with filter on txtype CreateMasternode', async () => {
+    const history = await client.account.listBurnHistory({
+      txtype: DfTxType.CREATE_MASTERNODE
+    })
+    expect(history.every(({ type }) => type === 'CreateMasternode')).toBeTruthy()
+  })
+
+  it('should listBurnHistory with filter on txtype CreateToken', async () => {
+    const history = await client.account.listBurnHistory({
+      txtype: DfTxType.CREATE_TOKEN
+    })
+    expect(history.every(({ type }) => type === 'CreateToken')).toBeTruthy()
+  })
+
+  it('should listBurnHistory with filter on token DFI', async () => {
+    const history = await client.account.listBurnHistory({
+      token: 'DFI'
+    })
+    expect(history.every(({ amounts }) => amounts[0].includes('DFI'))).toBeTruthy()
+  })
+
+  it('should listBurnHistory with filter on token GOLD', async () => {
+    const history = await client.account.listBurnHistory({
+      token: 'GOLD'
+    })
+    expect(history.every(({ amounts }) => amounts[0].includes('GOLD'))).toBeTruthy()
+  })
+
+  it('should listBurnHistory with maxBlockHeight 110', async () => {
+    const maxBlockHeight = 110
+    const history = await client.account.listBurnHistory({
+      maxBlockHeight
+    })
+    expect(history.every(({ blockHeight }) => blockHeight <= maxBlockHeight)).toBeTruthy()
+  })
+
+  it('should listBurnHistory with maxBlockHeight 500', async () => {
+    const maxBlockHeight = 500
+    const history = await client.account.listBurnHistory({
+      maxBlockHeight
+    })
+    expect(history.every(({ blockHeight }) => blockHeight <= maxBlockHeight)).toBeTruthy()
+  })
+
+  it('should listBurnHistory with depth 0', async () => {
+    const depth = 0
+    const history = await client.account.listBurnHistory()
+    const depthHistory = await client.account.listBurnHistory({
+      depth
+    })
+    const maxBlockHeight = Math.max(...history.map(el => el.blockHeight)) // Get maxBlockHeight from history
+    expect(depthHistory.every(({ blockHeight }) => blockHeight >= maxBlockHeight - depth)).toBeTruthy()
+  })
```

Comment:
hmm... TBH.. i don't quite understand the `depth` filter here

Reply:
Ok let me make that clearer